tests/run-pass/concurrency/locks.rs

```diff
@@ -0,0 +1,73 @@
+// ignore-windows
+
+use std::sync::{Arc, Mutex, RwLock};
+use std::thread;
+
+fn check_mutex() {
+    let data = Arc::new(Mutex::new(0));
+    let mut threads = Vec::new();
+
+    for _ in 0..3 {
+        let data = Arc::clone(&data);
+        let thread = thread::spawn(move || {
+            let mut data = data.lock().unwrap();
```

Comment:
Does this even lead to contention, or are threads typically done with their stuff before the next one tries to acquire the lock? Would it make sense to try and do some extra work in the critical section to make sure we actually get contention?

Reply:
With the current scheduler `thread::yield_now()` causes the next thread to be executed and be blocked on the lock. However, this is very fragile.